crates/iota-indexer/tests/rpc-tests/read_api.rs

```diff
@@ -1248,74 +1249,278 @@ fn try_get_past_object() {
+                    "Mismatch in latest_version for VersionTooHigh"
+                );
+            }
+            _ => panic!("Expected VersionTooHigh response, got: {:?}", result),
+        }
+    });
+}
+
+#[test]
+fn try_get_past_object_object_deleted() {
+    let ApiTestSetup {
+        runtime,
+        store,
+        client,
+        cluster,
+    } = ApiTestSetup::get_or_init();
+
+    runtime.block_on(async move {
+        indexer_wait_for_checkpoint(store, 1).await;
+
+        // Publish NFT package and create an NFT
+        let context = &cluster.wallet;
+        let (package_id, _, _) = publish_nfts_package(context).await;
+
+        let (sender, nft_object_id, tx_digest) = create_nft(context, package_id).await;
+
+        indexer_wait_for_transaction(tx_digest, store, client).await;
+
+        // Retrieve the latest object reference (which includes version) for deletion.
+        let nft_object_ref = cluster.get_latest_object_ref(&nft_object_id).await;
+
+        // Delete the NFT
+        let delete_tx = delete_nft(context, sender, package_id, nft_object_ref).await;
+        indexer_wait_for_transaction(delete_tx.digest, store, client).await;
+
+        assert!(
+            delete_tx.object_changes.is_some(),
+            "Object changes not found in delete transaction"
+        );
         assert!(
-            matches!(result, IotaPastObjectResponse::VersionTooHigh { .. }),
-            "expected VersionTooHigh response, got: {:?}",
-            result
+            delete_tx.object_changes.as_ref().unwrap().len() == 1,
+            "Object changes empty in delete transaction"
         );
+
+        // Retrieve the latest version after deletion.
+        let result = client
+            .try_get_object_before_version(nft_object_id, SequenceNumber::MAX)
+            .await
+            .expect("RPC call should succeed");
+
+        let deleted_version = match result {
+            IotaPastObjectResponse::ObjectDeleted(object_ref) => {
+                assert_eq!(
+                    object_ref.object_id, nft_object_ref.0,
+                    "Mismatch in ObjectDeleted object id"
+                );
+                assert_eq!(
+                    object_ref.version,
+                    nft_object_ref.1.next(),
+                    "Mismatch in ObjectDeleted object version"
+                );
+                assert_eq!(
+                    object_ref.digest,
+                    ObjectDigest::OBJECT_DIGEST_DELETED,
+                    "Mismatch in ObjectDeleted object digest"
+                );
+
+                object_ref.version
+            }
+            _ => panic!("Expected ObjectDeleted response, got: {:?}", result),
+        };
+
+        // Retrieve the latest object at the deleted version.
+        let result = client
+            .try_get_past_object(nft_object_id, deleted_version, None)
+            .await
+            .expect("RPC call should succeed");
+
+        match result {
```

Comment:
Would it work to write:
```
assert_eq!(result, IotaPastObjectResponse::ObjectDeleted(IotaObjectRef{
    object_id: nft_object_ref.0,
    version: nft_object_ref.1.next(),
    digest: ObjectDigest::OBJECT_DIGEST_DELETED,
}))
```
?

That was my original suggestion, that maybe got misinterpreted.